src/main/java/com/example/demo/dto/BookDto.java

```diff
@@ -1,14 +1,25 @@
 package com.example.demo.dto;
 
+import com.example.demo.annotation.unique.Unique;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Positive;
 import lombok.Data;
 import java.math.BigDecimal;
 
 @Data
 public class BookDto {
+    @NotBlank
+    @Positive
     private Long id;
```

Comment:
```suggestion
    @Positive
    private long id;
```
you can use NotNull with wrapper or just positive with primitive which can't be null